### PR TITLE
Changelog entry for v2.5.0

### PR DESCRIPTION
**Pull request by astorije**
~I need to do the intro blurb still, but it's late and there is already enough to review. I'll finish the blurb tomorrow (hence "Do not merge" label).~ **Done** (still Do not merge until todolist is cleared).

As usual, I tried to group PRs together when it made sense. For example, adding a new feature, then fixing bugs in multiple following PRs would end up on a single entry. Hope I missed nothing.
For the first time however, all lines were generated by a script I wrote (coming soon™) rather than built by hand, so risk of typos in the links is reduced by ∞%.

See [this](https://github.com/thelounge/lounge/blob/9bc314451159bf2a3e305bea92fa729d7f2638e9/CHANGELOG.md) for rendered version.

### TODO before merging / releasing

- [x] Date the changelog entry
- [x] Add `.lounge_home` to [`deb-lounge`](https://github.com/thelounge/deb-lounge)
- [x] Add `.lounge_home` to [`aur-lounge`](https://github.com/thelounge/aur-lounge)
- ~Add `.lounge_home` to [`docker-lounge`](https://github.com/thelounge/docker-lounge)~

### CODE REVIEW

**Comment by astorije** [comment]
Thanks for your tremendous help on this release @bews, @dgw, @eliemichel, @Gilles123, @iamstratos, @MaxLeiter, @PolarizedIons, @starquake, @xPaw, @YaManicKill, and @yashsriv!

File: CHANGELOG.md [hunk 1]
```diff
@@ -33,6 +33,139 @@ All sections are explained on the link above, they are all optional, and each of
...
+- Use `.lounge_home` to help distribution packages handle config paths right ([#1416](https://github.com/thelounge/lounge/pull/1416), [#1587](https://github.com/thelounge/lounge/pull/1587) by [@xPaw](https://github.com/xPaw), [#1418](https://github.com/thelounge/lounge/pull/1418) by [@astorije](https://github.com/astorije))
+- Implement push notifications ([#1124](https://github.com/thelounge/lounge/pull/1124), [#1445](https://github.com/thelounge/lounge/pull/1445), [#1572](https://github.com/thelounge/lounge/pull/1572), [#1468](https://github.com/thelounge/lounge/pull/1468) by [@xPaw](https://github.com/xPaw), [#1463](https://github.com/thelounge/lounge/pull/1463) by [@astorije](https://github.com/astorije))
+- Set default `/quit` message ([#1448](https://github.com/thelounge/lounge/pull/1448) by [@xPaw](https://github.com/xPaw))
+- Gracefully quit on <kbd>Ctrl</kbd> <kbd>C</kbd> ([#1477](https://github.com/thelounge/lounge/pull/1477) by [@xPaw](https://github.com/xPaw))
```
Comment (xPaw) [inline CHANGELOG.md]: Add a `+`

Comment (astorije) [reply CHANGELOG.md]: Yeah I started with that, then thought it was ugly so removed it. But least important detail in the world and fine either way so I re-added it 😅

**Comment by astorije** [comment]
Alright, I should be all set with the changelog! Let me know what you want me to amend :)

File: CHANGELOG.md [hunk 2]
```diff
@@ -33,6 +33,162 @@ All sections are explained on the link above, they are all optional, and each of
 
 <!-- New entries go after this line -->
 
+## v2.5.0 - 2017-10-@@@@@@
+
+For more details, [see the full changelog](https://github.com/thelounge/lounge/compare/v2.4.0...v2.5.0) and [milestone](https://github.com/thelounge/lounge/milestone/22?closed=1).
+
+Our [v2.3.0 release](https://github.com/thelounge/lounge/releases/tag/v2.3.0) was big, well, v2.5.0 is even bigger! 🎉
```
Comment (MaxLeiter) [inline CHANGELOG.md]: `If you thought the...`

Comment (MaxLeiter) [reply CHANGELOG.md]: broken sentence at the moment

File: CHANGELOG.md [hunk 3]
```diff
@@ -33,6 +33,162 @@ All sections are explained on the link above, they are all optional, and each of
...
+
+If you are a client user, you will first notice that The Lounge is now using your OS system font, then that status messages (joins, parts, nick changes, etc.) are now condensed with each other.
```
Comment (MaxLeiter) [inline CHANGELOG.md]: `If you are a client user, you will notice that The Lounge is now using your operating system's font, and that status messages (joins, parts, nick changes, etc.) are now condensed with each other.`

File: CHANGELOG.md [hunk 4]
```diff
@@ -33,6 +33,162 @@ All sections are explained on the link above, they are all optional, and each of
...
+
+<p align="center">
+  <img width="555" alt="Condensed status messages displaying 3 users have joined the channel, 4 users have quit, and 1 user has changed nick" src="https://user-images.githubusercontent.com/113730/31530599-a6d719f2-afaf-11e7-9b82-db0365c41c4e.png">
+  <br>
+  <em>Condensed status messages</em>
+</p>
+
+After a while, you'll realize that the client now automatically reconnects when losing network (farewell, `Client connection lost — Click here to reconnect`!), and that channel backlog now automatically loads when scrolling. Unfortunately, that last bit forced us to phase out the `/clear` command for technical reasons.
```
Comment (MaxLeiter) [inline CHANGELOG.md]: `...client now automatically reconnects when you lose your network connection`

`and that channel backlog` -> `and that the channel backlog now automatically loads when you scroll up`

File: CHANGELOG.md [hunk 5]
```diff
@@ -33,6 +33,162 @@ All sections are explained on the link above, they are all optional, and each of
...
+
+If you go to the _Settings_ page, you'll notice that The Lounge now supports push notifications (yes, even on mobile, except iOS), and that you can now access information about open sessions and even remotely log them out.
```
Comment (MaxLeiter) [inline CHANGELOG.md]: `and that you can now access information about open sessions and even remotely log them out.` -> `and that you can now access information about open sessions and you can even remotely log them out.`

File: CHANGELOG.md [hunk 6]
```diff
@@ -33,6 +33,162 @@ All sections are explained on the link above, they are all optional, and each of
...
+
+<p align="center">
+  <img width="477" alt="Session list with the ability to disconnect remote sessions" src="https://user-images.githubusercontent.com/113730/31530598-a6c86b46-afaf-11e7-9272-eb742d328686.png">
+  <br>
+  <em>Current and remote sessions</em>
+</p>
+
+As a server administrator, you might be interested by a few new additions: themes can now be [retrieved from npm](https://www.npmjs.com/search?q=keywords%3Athelounge-theme) and installed using a new CLI command `lounge install <theme-name>`; integration with LDAP has been completely refactored; The Lounge can now be bound to Unix sockets.
```
Comment (MaxLeiter) [inline CHANGELOG.md]: As a server administrator, you might be interested by a few new additions: 


- themes can now be retrieved from npm and installed using a new CLI command (`lounge install <theme-name>`)
- integration with LDAP has been completely refactored
- The Lounge can now be bound to Unix sockets

File: CHANGELOG.md [hunk 7]
```diff
@@ -33,6 +33,162 @@ All sections are explained on the link above, they are all optional, and each of
...
+
+⚠️ Note that `--home` is now deprecated in favor of the `$LOUNGE_HOME` environment variable (or `.lounge_home` file in the installation directory). Also, if you are running The Lounge behind a proxy (like nginx or Apache), you will need to make sure that `reverseProxy` is set to `true` and the `X-Forwarded-For` correctly set for session listing to work correctly on the client.
```
Comment (MaxLeiter) [inline CHANGELOG.md]: `or .lounge_home\ file` -> `or the .lounge_home\ file`

Comment (MaxLeiter) [reply CHANGELOG.md]: `the X-Forwarded-For` -> `the X-Forwarded-For header` is

File: CHANGELOG.md (hunk 6 above)
Comment (MaxLeiter) [reply CHANGELOG.md]: in a few new additions*

**Comment by astorije** [comment]
@MaxLeiter approved the changes to me on IRC.

**Pull request merged**